crypto/fipsmodule/rand/entropy/entropy_sources.c

```diff
@@ -7,34 +7,52 @@
 #include <openssl/base.h>
 
 #include "internal.h"
+#include "../internal.h"
+#include "../../delocate.h"
 
-static int fake_void(void) {
+static int entropy_default_initialize(void) {
   return 1;
 }
 
-static int fake_rand(uint8_t a[CTR_DRBG_ENTROPY_LEN]) {
-  OPENSSL_cleanse(a, CTR_DRBG_ENTROPY_LEN);
-  return 1;
+static void entropy_default_cleanup(void) {
 }
 
-static int fake_rand_(uint8_t a[RAND_PRED_RESISTANCE_LEN]) {
-  OPENSSL_cleanse(a, RAND_PRED_RESISTANCE_LEN);
+static int entropy_default_get_seed(uint8_t seed[CTR_DRBG_ENTROPY_LEN]) {
+  CRYPTO_sysrand_for_seed(seed, CTR_DRBG_ENTROPY_LEN);
   return 1;
 }
 
-int get_entropy_source(struct entropy_source *entropy_source) {
-
-  // In the future this function will lazily initialise a global entropy source.
+static int entropy_default_get_prediction_resistance(
+  uint8_t pred_resistance[RAND_PRED_RESISTANCE_LEN]) {
+  if (have_fast_rdrand() != 1 ||
+      rdrand(pred_resistance, RAND_PRED_RESISTANCE_LEN) != 1) {
+    return 0;
+  }
+  return 1;
+}
 
-  GUARD_PTR(entropy_source);
+static int entropy_default_randomize(void) {
```

Comment:
just curious, what's this function meant to do?

Reply:
I model the "entropy source" as a stateful object and I (might) need a function that can request a randomization of the object.

I do that because I want to implement a tree-DRBG that has a live entropy source at it's root. But this means that to handle uniqueness-breaking events (e.g. a fork), we need a method to randomize the tree-DRBG. This function is intended to do that.

However, I might decide later that this function is actually redundant and the entropy source backend must handle this UBEs by its own.